src/dashboard/Data/Browser/Browser.react.js

```diff
@@ -1255,15 +1257,12 @@ class Browser extends DashboardView {
     });
   }
 
-  async confirmExportSelectedRows(rows) {
-    this.setState({ rowsToExport: null });
+  async confirmExportSelectedRows(rows, type, indentation) {
+    this.setState({ rowsToExport: null, exporting: true, exportingCount: 0 });
     const className = this.props.params.className;
     const query = new Parse.Query(className);
 
-    if (rows['*']) {
-      // Export all
-      query.limit(10000);
-    } else {
+    if (!rows['*']) {
```

Comment:
What limit if any is applied if `rows['*'] == true`?

Reply:
`query.limit(objectIds.length);`

Reply:
But that's in the if block:

https://github.com/parse-community/parse-dashboard/blob/46b1346f20e01d28e772e5ef8e1706c8ce63e917/src/dashboard/Data/Browser/Browser.react.js#L1265-L1273

What happens if the condition is false, which limit is then applied?

Reply:
Oh right, it uses query.eachBatch, which does not accept a limit (so unlimited)

Reply:
So it will export all objects of a class?

Reply:
If `rows["*"]` is export all, so the inverse is selected rows only

Reply:
Can you help understand the logic of that if block. It seems to me it means if only selected rows should be exported (!rows[“*”]), then set the limit in the block. Inversely it implies that if all rows of a class should be exported, then do not set any limit.

If that’s how it works, then
- why do we need to set a limit at all if the limit can already be inferred from the selected objectIds? In other words, if object A and B are selected for export, then we’ll never export more than 2 objects, so why do we need a limit?
- if all rows of a class should be exported (no rows selected), then there is no limit set. If a class has a high number of records, then that can have unintended consequences, i.e bring down the DB or cause high data traffic costs

Reply:
- the limit code is existing from the previous behaviour of export selected, I’m assuming it’s to override any server default limit. 
- it’s not possible to limit `eachBatch`, it automatically iterates across the whole class.

Reply:
> I’m assuming it’s to override any server default limit

makes sense

> it’s not possible to limit eachBatch, it automatically iterates across the whole class

I think such a query requires some kind of safety confirmation.

Reply:
How about a "confirm export all" field and a warning that the function can cause database slowness?